argo lint panics on a certain invalid workflow

## Summary

What happened/what you expected to happen?
`argo lint workflow.yaml` panics.

This is not a blocker for me. Just stumbled upon it and decided to report.

What version of Argo Workflows are you running?
`v3.3.0-rc2`

## Diagnostics

Either a workflow that reproduces the bug, or paste you whole workflow YAML, including status, something like:

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Workflow
spec:
  entrypoint: root
  templates:
    - name: template
      inputs:
        parameters:
          - name: data
      container:
        name: main
        image: alpine
    - name: root
      inputs:
        parameters:
          - name: anything_param
      dag:
        tasks:
          - name: task
            template: template
            arguments:
              parameters:
                - name: data
                  # There is a bug here. I still keep forgetting which syntax is right >_<
                  valueFrom:
                    parameter: "{{inputs.parameters.anything_param}}"
  arguments:
    parameters:
      - name: anything_param
        value: anything_param
```

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x19c6f3b]

goroutine 1 [running]:
github.com/argoproj/argo-workflows/v3/workflow/validate.validateDAGTaskArgumentDependency({{0xc000189080, 0x1, 0x4}, {0x0, 0x0, 0x0}}, {0x478a408, 0x0, 0x23ec660})
        /home/runner/work/argo-workflows/argo-workflows/workflow/validate/validate.go:1332 +0x3bb
github.com/argoproj/argo-workflows/v3/workflow/validate.(*templateValidationCtx).validateDAG(0xc00012bb40, 0xc00042dd40, 0x2526839, 0xc000180900)
        /home/runner/work/argo-workflows/argo-workflows/workflow/validate/validate.go:1311 +0xce5
github.com/argoproj/argo-workflows/v3/workflow/validate.(*templateValidationCtx).validateTemplate(0xc00012bb40, 0xc000180240, 0xc00012e9c0, {0x32dd130, 0xc00042dbf0})
        /home/runner/work/argo-workflows/argo-workflows/workflow/validate/validate.go:399 +0xf3c
github.com/argoproj/argo-workflows/v3/workflow/validate.(*templateValidationCtx).validateTemplateHolder(0x21628c0, {0x32ec2c0, 0xc00012e9c0}, 0xf, {0x32dd130, 0xc00042dbf0})
        /home/runner/work/argo-workflows/argo-workflows/workflow/validate/validate.go:486 +0x2e9
github.com/argoproj/argo-workflows/v3/workflow/validate.ValidateWorkflow({0x32bb620, 0xc000b86f50}, {0x32bb600, 0xc000b86f70}, 0xc000264900, {0x1, {0x0, 0x0}, 0x0, 0x0, ...})
        /home/runner/work/argo-workflows/argo-workflows/workflow/validate/validate.go:208 +0xfd6
github.com/argoproj/argo-workflows/v3/server/workflow.(*workflowServer).LintWorkflow(0xc000990b40, {0x330f6b0, 0xc0008bbe90}, 0xc00012bb00)
        /home/runner/work/argo-workflows/argo-workflows/server/workflow/workflow_server.go:504 +0x1d5
        /home/runner/work/argo-workflows/argo-workflows/pkg/apiclient/argo-kube-workflow-service-client.go:92 +0x2b
github.com/argoproj/argo-workflows/v3/pkg/apiclient.(*errorTranslatingWorkflowServiceClient).LintWorkflow(0x250efd0, {0x330f6b0, 0xc0008bbe90}, 0xc000264900, {0x0, 0x400, 0x32bb100})
        /home/runner/work/argo-workflows/argo-workflows/pkg/apiclient/error-translating-workflow-service-client.go:85 +0x33
github.com/argoproj/argo-workflows/v3/cmd/argo/lint.lintData({0x330f6b0, 0xc0008bbe90}, {0xc00005e090, 0x8c}, {0xc00083ec00, 0x2b0, 0x400}, 0xc000a27be8)
        /home/runner/work/argo-workflows/argo-workflows/cmd/argo/lint/lint.go:224 +0x883
        /home/runner/work/argo-workflows/argo-workflows/cmd/argo/lint/lint.go:155 +0x2aa
path/filepath.walk({0xc00005e090, 0x8c}, {0x3333620, 0xc000483c70}, 0xc0009dda80)
        /opt/hostedtoolcache/go/1.17.6/x64/src/path/filepath/path.go:418 +0x125
path/filepath.Walk({0xc00005e090, 0x8c}, 0xc0009dda80)
        /opt/hostedtoolcache/go/1.17.6/x64/src/path/filepath/path.go:505 +0x6c
github.com/argoproj/argo-workflows/v3/cmd/argo/lint.Lint({0x330f6b0, 0xc0008bbe90}, 0xc000a27be8)
        /home/runner/work/argo-workflows/argo-workflows/cmd/argo/lint/lint.go:124 +0x1eb
github.com/argoproj/argo-workflows/v3/cmd/argo/lint.RunLint({0x330f6b0, 0xc0008bbe90}, {0x3332660, 0xc0005b5d20}, {0xc0000d2600, 0x4, 0x4}, {0x250bd5c, 0x6}, 0x0, ...)
        /home/runner/work/argo-workflows/argo-workflows/cmd/argo/lint/lint.go:98 +0x29f
github.com/argoproj/argo-workflows/v3/cmd/argo/commands.NewLintCommand.func1(0xc000833680, {0xc00094e4d0, 0x1, 0x1})
        /home/runner/work/argo-workflows/argo-workflows/cmd/argo/commands/lint.go:52 +0x325
github.com/spf13/cobra.(*Command).execute(0xc000833680, {0xc00094e4a0, 0x1, 0x1})
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:860 +0x5f8
github.com/spf13/cobra.(*Command).ExecuteC(0xc000832c80)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:902
main.main()
        /home/runner/work/argo-workflows/argo-workflows/cmd/argo/main.go:14 +0x1e
```

---
<!-- Issue Author: Don't delete this message to encourage other users to support your issue! -->
**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

--- a/workflow/validate/validate.go
+++ b/workflow/validate/validate.go
@@ -1326,6 +1326,9 @@ func validateDAGTaskArgumentDependency(arguments wfv1.Arguments, ancestry []stri
 	}
 
 	for _, param := range arguments.Parameters {
+		if param.Value == nil {
+			return errors.Errorf(errors.CodeBadRequest, "missing value for parameter '%s'", param.Name)
+		}
 		if strings.HasPrefix(param.Value.String(), "{{tasks.") {
 			// All parameter values should have been validated, so
 			// index 1 should exist.

--- a/workflow/validate/validate_dag_test.go
+++ b/workflow/validate/validate_dag_test.go
@@ -1033,3 +1033,42 @@ func TestDAGOutputsResolveTaskAggregatedOutputs(t *testing.T) {
 	_, err := validate(dagOutputsResolveTaskAggregatedOutputs)
 	assert.NoError(t, err)
 }
+
+var dagMissingParamValueInTask = `
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+spec:
+  entrypoint: root
+  templates:
+    - name: template
+      inputs:
+        parameters:
+          - name: data
+      container:
+        name: main
+        image: alpine
+    - name: root
+      inputs:
+        parameters:
+          - name: anything_param
+      dag:
+        tasks:
+          - name: task
+            template: template
+            arguments:
+              parameters:
+                - name: data
+                  valueFrom:
+                    parameter: "{{inputs.parameters.anything_param}}"
+  arguments:
+    parameters:
+      - name: anything_param
+        value: anything_param
+`
+
+func TestDAGMissingParamValueInTask(t *testing.T) {
+	_, err := validate(dagMissingParamValueInTask)
+	if assert.NotNil(t, err) {
+		assert.Contains(t, err.Error(), "templates.root.tasks.task missing value for parameter 'data'")
+	}
+}

